src/Symfony/Component/Form/DependencyInjection/FormPass.php

```diff
@@ -46,29 +50,36 @@ public function process(ContainerBuilder $container)
         }
 
         $definition = $container->getDefinition($this->formExtensionService);
```

Comment:
Ok, thank you for the demonstration ;), it indeed looks like this extra check is not needed.